api/app/files/api.py

```diff
@@ -45,9 +46,33 @@ def get(self):
             return send_file(temp.name, as_attachment=True, attachment_filename=args['filename'], mimetype='application/pdf')
 
 
+
+class FileAPIImproved(FileUploadMixin, restful.Resource):
+    def get(self, file_id):
+        # TO DO: Query the File table
+
+        req_parser = reqparse.RequestParser()
+        req_parser.add_argument('filename', type=str, required=True)

```

Comment:
So over here this should be `req_parser.add_argument('file_id', type=int, required=True)`